### DISCUSSION TITLE
Stuck on HTML Foundry Fund Me

### DISCUSSION THREAD

**Discussion by handychristian** [Cyfrin/foundry-full-course-cu#2379] [General]
Hey why its rpc url error?, i already followed the steps from connect to import the anvil private key account and theres a balance, but why i cant use fundme button, at the video says if rpc url error because the account doesnt have balance 

![image](https://github.com/user-attachments/assets/7c9648f0-4f51-4d52-a2bd-c57cef9e92af)
![image](https://github.com/user-attachments/assets/f1fcc1f2-1d3b-49d9-be94-2aa8f6addf0b)

**Comment by EngrPips** [comment]
Do you have an Anvil chain running?

**Reply by handychristian** [reply]
yes its running and as you see its connected at top right

**Reply by handychristian** [reply]
i searched the issue on chatgpt this is his answer what do you think?

![image](https://github.com/user-attachments/assets/3c80ed1d-e06a-4b31-a378-dbfeae86b0ca)

**Reply by urosognjenovic** [reply]
The network is set to Anvil in Metamask, but that doesn't mean that your local Anvil node is running.

**Reply by handychristian** [reply]
![image](https://github.com/user-attachments/assets/e855e2e9-4497-45d7-b6cd-12a741f81a5d)
is this local anvil node?

**Reply by EngrPips** [reply]
Yes.

**Reply by handychristian** [reply]
so local anvil is running, why its still error?

**Reply by EngrPips** [reply]
Did you correctly set the `rpc-url` for the anvil chain on metamask?

**Reply by handychristian** [reply]
yes i set it right
![Uploading image.png…]()

**Reply by EngrPips** [reply]
The image is not loading for me.

**Comment by ren06** [comment]
I am having the exact same issue

**Reply by EngrPips** [reply]
I am unsure why you are running into this error but that was a quick section and i will not sweat too much over it. Also I will revisit that section as soon as i am chanced and see what might be wrong.

**Reply by yawarasuuna** [reply]
> I am having the exact same issue

I had the same problem while using Firefox, so I moved to a chromium based web browser, and it worked.


edit: I just noticed OP print screen is already using chrome, so it might not be applicable to them. I'll leave this up in case someone faces the same problem as I did

**Comment by EngrPips** [comment]
Interesting.

**Comment by mirkolugano1** [comment]
I am having a similar issue here.
I am using Chrome on Windows, and have spun up Anvil in WSL Ubuntu. I deployed the contract using the DeployFundMe.sol script, gave my Metamask account some ETH from an Anvil account so that it could have funds to send, I click the "fund" button, Metamask pops up, I confirm the transaction but Metamask says "Transaction failed: Internal JSON RPC error". Funny enough, my Chrome Console does not show anything, not even the normal console.log() data from within the "normal" website execution. I see that the transaction is being sent to Anvil correctly, since on the Anvil tab in my Terminal I see this right popping up after I hit "fund"

eth_call
eth_blockNumber
eth_sendRawTransaction
eth_call
eth_call

but still, the transaction fails and I can see no further details about it, neither in Chrome nor in Anvil.
I will investigate a bit further but I don't plan to spend too much time on this. If somebody has ideas or a quick fix, you're welcome.

**Reply by EngrPips** [reply]
Tough to say what was wrong. I did this section without any issues at all. Maybe you should add an `Anvil` account to your Metamask and then connect that account to run your process instead of transferring funds.

**Reply by mirkolugano1** [reply]
thanx for the reply - I tried that but still got the same error. Must be something with my browser I suppose. I'll just ignore it for the moment and go on with the rest of the course.

**Reply by EngrPips** [reply]
That is tough, Yeah don't let that block you.
